### ISSUE TITLE
Dict ends up dumping the whole dict whenever it is displayed as a member of another type

### ISSUE THREAD

**Issue by denizyuret** [JuliaLang/julia#13710]
If you refer to a Dict in repl, or call display on it, it will nicely get shortened.  However if you have a Type that has a Dict member, the whole dictionary gets dumped when displaying an instance of that Type.  A similar problem does not occur in regular Arrays.

**Comment by quinnj** [comment]
A reproducible output:
```julia
type TestDict
    a::Int
    b::Dict
    c::Array
end

array = Pair[i=>"$i" for i = 1:200]
dict = Dict(array)

TestDict(1,dict,array)
```
With output from the last statement:
```julia
julia> TestDict(1,dict,array)
TestDict(1,Dict(11=>"11",134=>"134",158=>"158",160=>"160",29=>"29",131=>"131",173=>"173",74=>"74",176=>"176",57=>"57",31=>"31",70=>"70",33=>"33",114=>"114",165=>"165",96=>"96",133=>"133",49=>"49",84=>"84",117=>"117",93=>"93",50=>"50",77=>"77",80=>"80",188=>"188",132=>"132",195=>"195",157=>"157",153=>"153",79=>"79",71=>"71",120=>"120",154=>"154",76=>"76",184=>"184",167=>"167",122=>"122",143=>"143",62=>"62",107=>"107",169=>"169",21=>"21",10=>"10",102=>"102",180=>"180",162=>"162",129=>"129",88=>"88",149=>"149",5=>"5",63=>"63",41=>"41",15=>"15",148=>"148",55=>"55",66=>"66",155=>"155",150=>"150",95=>"95",139=>"139",90=>"90",111=>"111",14=>"14",175=>"175",200=>"200",135=>"135",125=>"125",172=>"172",146=>"146",130=>"130",17=>"17",65=>"65",44=>"44",37=>"37",192=>"192",110=>"110",127=>"127",13=>"13",99=>"99",137=>"137",141=>"141",105=>"105",30=>"30",1=>"1",47=>"47",91=>"91",43=>"43",9=>"9",128=>"128",147=>"147",104=>"104",183=>"183",103=>"103",152=>"152",38=>"38",118=>"118",48=>"48",100=>"100",81=>"81",140=>"140",196=>"196",19=>"19",51=>"51",191=>"191",177=>"177",123=>"123",163=>"163",108=>"108",164=>"164",2=>"2",89=>"89",197=>"197",136=>"136",151=>"151",144=>"144",142=>"142",78=>"78",159=>"159",112=>"112",106=>"106",121=>"121",109=>"109",182=>"182",26=>"26",138=>"138",4=>"4",168=>"168",94=>"94",115=>"115",54=>"54",40=>"40",101=>"101",171=>"171",60=>"60",34=>"34",178=>"178",198=>"198",113=>"113",12=>"12",189=>"189",16=>"16",92=>"92",72=>"72",28=>"28",97=>"97",68=>"68",124=>"124",39=>"39",46=>"46",85=>"85",170=>"170",25=>"25",42=>"42",181=>"181",58=>"58",193=>"193",59=>"59",8=>"8",199=>"199",185=>"185",20=>"20",194=>"194",52=>"52",18=>"18",69=>"69",166=>"166",187=>"187",35=>"35",83=>"83",64=>"64",145=>"145",45=>"45",86=>"86",67=>"67",126=>"126",32=>"32",190=>"190",82=>"82",7=>"7",75=>"75",87=>"87",179=>"179",3=>"3",61=>"61",36=>"36",98=>"98",174=>"174",161=>"161",186=>"186",22=>"22",6=>"6",24=>"24",73=>"73",119=>"119",156=>"156",53=>"53",116=>"116",23=>"23",27=>"27",56=>"56"),Pair[1=>"1",2=>"2",3=>"3",4=>"4",5=>"5",6=>"6",7=>"7",8=>"8",9=>"9",10=>"10"  …  191=>"191",192=>"192",193=>"193",194=>"194",195=>"195",196=>"196",197=>"197",198=>"198",199=>"199",200=>"200"])
```
(notice the `...` for the `Vector{Pair}` whereas the Dict is displayed in full).

**Comment by stevengj** [comment]
Hopefully will be fixed by #14052 ... the REPL will `show` with `verbosity=1` attached to the `io` object, which will get passed down recursively when showing members of types.
